### DISCUSSION TITLE
How to add --sync-shared-folders --single-directory to docker

### DISCUSSION THREAD

**Discussion by ukreddy-erwin** [abraunegg/onedrive#1549] [Q&A] [answered]
I am using below command for onedrive shared folders.
```
./onedrive --synchronize --verbose --sync-shared-folders --single-directory 'Jenkins' --resync
```
This is used to download only a particular folder.

How to use the similar command with docker?

<html>
<body>
<p style="box-sizing: border-box; margin-bottom: 16px; margin-top: 0px !important; color: rgb(36, 41, 46); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">You can provided multiple options as environment variables to docker.</p>

ONEDRIVE_UID | UserID (UID) to run as | 1000
-- | -- | --
ONEDRIVE_GID | GroupID (GID) to run as | 1000
ONEDRIVE_VERBOSE | Controls "--verbose" switch on onedrive sync. Default is 0 | 1
ONEDRIVE_DEBUG | Controls "--verbose --verbose" switch on onedrive sync. Default is 0 | 1
ONEDRIVE_DEBUG_HTTPS | Controls "--debug-https" switch on onedrive sync. Default is 0 | 1
ONEDRIVE_RESYNC | Controls "--resync" switch on onedrive sync. Default is 0 | 1
ONEDRIVE_DOWNLOADONLY | Controls "--download-only" switch on onedrive sync. Default is 0 | 1
ONEDRIVE_LOGOUT | Controls "--logout" switch. Default is 0 | 1

<p style="box-sizing: border-box; margin-bottom: 16px; margin-top: 0px; color: rgb(36, 41, 46); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">But these two options are not available.</p><p style="box-sizing: border-box; margin-bottom: 16px; margin-top: 0px; color: rgb(36, 41, 46); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">--sync-shared-folders --single-directory 'Jenkins'</p><p style="box-sizing: border-box; margin-bottom: 0px !important; margin-top: 0px; color: rgb(36, 41, 46); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">How to pass them to docker container?</p>
</body>
</html>

**Comment by abraunegg** [comment, marked as answer]
@ukreddy-erwin 
> But these two options are not available.
> 
> --sync-shared-folders --single-directory 'Jenkins'

To support using `--sync-shared-folders` in a Docker container should be possible, as this is a binary 0|1 option that follows the others that are allowed as a passed in option. Please raise a feature request to have this added.

This being said, as per the documentation for Shared Business Folders, you can enable | disable the syncing of Shared Business Folders via the configuration file by modifying the appropriate value.

You can also edit the Docker configuration file by inspecting the Docker volume and editing the configuration that it runs

Please read the following documentation:
* https://github.com/abraunegg/onedrive/blob/master/docs/BusinessSharedFolders.md
* https://github.com/abraunegg/onedrive/blob/master/docs/Docker.md


The support of `--single-directory` option is not possible. This parameter is not compatible with Docker.

**Comment by ukreddy-erwin** [comment]
I added business_shared_folders and sync_list file in the docker volume assigned for config files. 
```
docker volume inspect onedrive_conf
```
And when I connect to the docker container and ran display config, it is not showing those configurations.
```
export ONEDRIVE_DATA_DIR="${HOME}/OneDrive"
mkdir -p ${ONEDRIVE_DATA_DIR}
docker run -it --name onedrive -v onedrive_conf:/onedrive/conf -v "${ONEDRIVE_DATA_DIR}:/onedrive/data" -e "ONEDRIVE_RESYNC:1" -e "ONEDRIVE_VERBOSE:1" -e "ONEDRIVE_DEBUG:1" driveone/onedrive:latest
```
I understood as there is no --sync-shared-folders option provided in the container, it won't sync the files. But it should display the configuration right?

**Reply by abraunegg** [reply]
@ukreddy-erwin 
> But it should display the configuration right?

Correct it should .. but did you edit the configuration files correctly?

What is the explicit contents of the following files in your onedrive_conf volume:
* business_shared_folders
* sync_list
* config

**Reply by ukreddy-erwin** [reply]
I didn't add config file as it should pick the default one. 

**business_shared_folders**
```
Jenkins
```
**sync_list**
```
Jenkins/LatestBuilds
```

**Reply by abraunegg** [reply]
@ukreddy-erwin 
> I didn't add config file as it should pick the default one

Please can you think this through a little more ..

If you are using the default one, it is using the application defaults. You need to modify the application config so that performs the operations you are asking it to.

So of course, because of your own actions, the application is not displaying the configuration you 'think' it should be ....

**Reply by ukreddy-erwin** [reply]
I mean, I edited the default config file to add
Business Shared Folders configured     = true

**Reply by ukreddy-erwin** [reply]
And with the direct onedrive instead of docker, without the config file modification, just with business_shared_folders and sync_list, it is picking the display config directly.

**Reply by abraunegg** [reply]
@ukreddy-erwin 
> I mean, I edited the default config file to add
> Business Shared Folders configured = true

This is not correct.

Please read the documentation.

**Reply by abraunegg** [reply]
@ukreddy-erwin 
> And with the direct onedrive instead of docker, without the config file modification, just with business_shared_folders and sync_list, it is picking the display config directly.

You need to edit the configuration files correctly. Please read the documentation.

Once this is done, then you probably need to 'restart' the Docker container to pick up the new configuration.

This is all explained in the documentation.

Please read the documentation.

**Reply by ukreddy-erwin** [reply]
In this document: https://github.com/abraunegg/onedrive/blob/master/docs/BusinessSharedFolders.md

you created  ~/.config/onedrive/business_shared_folders file only.

As you are running --sync-shared-folders, may be it doesn't require separate configuration.
 onedrive --synchronize --sync-shared-folders --verbose

Does keeping the below line in the same folder with file (config) will suffice?
sync_business_shared_folders = "true"

I didn't find option for --single-directory

**Reply by abraunegg** [reply]
@ukreddy-erwin 
As I have indicated above: The support of `--single-directory` option is not possible. This parameter is not compatible with Docker.
